launcher/launcher.go

```diff
@@ -179,3 +254,327 @@ func (launcher *Launcher) RuntimeStatusChannel() <-chan RuntimeStatus {
 /***********************************************************************************************************************
  * Private
  **********************************************************************************************************************/
+
+func (launcher *Launcher) handleRunnerStatus(ctx context.Context) {
+	for {
+		select {
+		case instances := <-launcher.instanceRunner.InstanceStatusChannel():
+			launcher.updateInstancesStatuses(instances)
+
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func (launcher *Launcher) updateInstancesStatuses(instances []runner.InstanceStatus) {
+	launcher.runMutex.Lock()
+	defer launcher.runMutex.Unlock()
+
+	updateInstancesStatus := &UpdateInstancesStatus{Instances: make([]cloudprotocol.InstanceStatus, 0, len(instances))}
+
+	for _, instanceStatus := range instances {
+		currentInstance, ok := launcher.currentInstances[instanceStatus.InstanceID]
+		if !ok {
+			log.WithField("instanceID", instanceStatus.InstanceID).Warn("Not running instance status received")
+			continue
+		}
+
+		if currentInstance.runStatus.State != instanceStatus.State {
+			currentInstance.setRunStatus(instanceStatus)
+
+			if !launcher.runInstancesInProgress {
+				updateInstancesStatus.Instances = append(updateInstancesStatus.Instances,
+					currentInstance.getCloudStatus())
+			}
+		}
+	}
+
+	if len(updateInstancesStatus.Instances) > 0 {
+		launcher.runtimeStatusChannel <- RuntimeStatus{UpdateStatus: updateInstancesStatus}
+	}
+}
+
+func (launcher *Launcher) createNewInstance(instanceIdent cloudprotocol.InstanceIdent) (InstanceInfo, error) {
+	instance := InstanceInfo{
+		InstanceIdent: instanceIdent,
+		InstanceID:    uuid.New().String(),
+		UnitSubject:   launcher.isCurrentSubject(instanceIdent.SubjectID),
+	}
+
+	if err := launcher.storage.AddInstance(instance); err != nil {
+		return instance, aoserrors.Wrap(err)
+	}
+
+	return instance, nil
+}
+
+func (launcher *Launcher) updateRunningFlags(runInstances []InstanceInfo) error {
+	// Clear running flag for not running instances
+	currentRunInstances, err := launcher.storage.GetRunningInstances()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+currentLoop:
+	for _, currentInstance := range currentRunInstances {
+		for _, runInstance := range runInstances {
+			if currentInstance.InstanceIdent == runInstance.InstanceIdent {
+				continue currentLoop
+			}
+		}
+
+		currentInstance.Running = false
+
+		if err := launcher.storage.UpdateInstance(currentInstance); err != nil {
+			return aoserrors.Wrap(err)
+		}
+	}
+
+	// Set running flag for running instances
+	for _, runInstance := range runInstances {
+		if !runInstance.Running {
+			runInstance.Running = true
+
+			if err := launcher.storage.UpdateInstance(runInstance); err != nil {
+				return aoserrors.Wrap(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (launcher *Launcher) runInstances(runInstances []InstanceInfo) {
```

Comment:
Could you rename the method? Because this function also stops not actual instances.

Reply:
It starts desired instances and stops others. It takes instances to be run as parameter. Whar would be more suitable name: `processInstancess`? any suggestion?

Reply:
ok